src/prototype_decryption_encryption/cuban_crow.py

```diff
@@ -39,11 +55,29 @@ def interrogation_room_upload(file_location: str, checksum: str):
 
 def encryption_key_store_upload(file_part: bytes) -> Tuple[str, str, int]:
     """
-    TODO:
-    Implement based on requirements in
-    Prototype Script 2/3: Encryption Key Store (Upload) GDEV-1239
+    Encryption key store functionality:
+    Extract header envelope from the first file part
+    Decrypt header & extract key
+    Return key, key id and offset
     """
-    return "", "", 0
+
+    file_stream = io.BytesIO(file_part)
+
+    # request crypt4gh private key
+    receiver_sec = request_cryp4gh_private_key()
+    receiver_keys = [(0, receiver_sec, None)]
+
+    session_keys, __ = crypt4gh.header.deconstruct(
+        file_stream,
+        keys=receiver_keys,
+    )
```

Comment:
```suggestion
    ghga_sec = request_cryp4gh_private_key()
    keys = [(0, ghga_sec, None)]

    session_keys, __ = crypt4gh.header.deconstruct(
        file_stream,
        keys=keys,
    )
```
We already changed the key names, so just keep that in line in the code